### PR TITLE
212 implement menu status selector

### PR DESCRIPTION
**Pull request by sepowitz**
This selector will be useful on any frontend using b-dux to determine the status of the requested menu, and to give some indication to the user when they are ordering for. 

I added a lot of comments, so please read through them.

### CODE REVIEW

File: tests/selectors/menuStatusForOrder.test.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        testLocation,
+        `current_daypart.${testServiceType}.is_orderable`,
+      ),
+      currentDaypartIsInTheFuture:
+        get(testLocation, `current_daypart.${testServiceType}.is_orderable`) &&
+        !get(testLocation, `current_daypart.${testServiceType}.is_current`),
+    });
+  });
+
+  it('returns correct response when wantsFutureOrder is false, and validOrderTimeForOrder cannot be found', () => {
```
Comment (hhff) [inline tests/selectors/menuStatusForOrder.test.js]: I expected these tests to be more like:

- "When a user started an order yesterday, but came back to the website today"
- "When a user has a catering order, but opened the website when the store is currently closed"

Have you got any plans to add real world scenarios? I can't really reason about wether this code covers all the use cases we discussed without them

**Comment by sepowitz** [comment]
@joshiefishbein 

You'll notice we're passing additional arguments into these selectors -- in order to make them more testable -- which is not memoized by reselect out of the box. Instead, they recommend using lodash (or any other memoization function) to memoize selectors that get passed additional arguments outside or state/props.

File: src/selectors/orders/menuStatusForOrder.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        };
+      }
+    }),
+);
+
+export const menuStatusForOrder = createSelector(state =>
+  _menuStatusForOrder(state)(
+    validOrderTimeForOrder(state)(
+      luxonDateTimeFromRequestedAt(
+        get(state, 'session.order.orderData.requested_at'),
```
Comment (joshiefishbein) [inline src/selectors/orders/menuStatusForOrder.js]: Can't wait for a pipe operator for JS. I've used [Lodash's Flow util](https://lodash.com/docs/4.17.11#flow) in other situations where I've been forced to read functions inside-out, you might like it for moments like this.

File: src/selectors/orders/menuStatusForOrder.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ const {
   ASAP_ORDER_REQUEST,
   INVALID_REQUESTED_AT,
   REQUESTED_AT_HAS_PASSED,
-  ORDERING_FOR_CURRENT_DAYPART,
+  ORDERING_FOR_FIRST_AVAILABLE_DAYPART,
```
Comment (hhff) [inline src/selectors/orders/menuStatusForOrder.js]: wouldn't it be more correct to say: "ORDERING_FOR_FIRST_AVAILABLE_VALID_TIME"

File: src/selectors/orders/menuStatusForOrder.js (hunk 2 above)
Comment (sepowitz) [reply src/selectors/orders/menuStatusForOrder.js]: def more readable, but doesn't seem necessary as a dependency just for this one selector. I'm pumped for some pipes too!

**Pull request merged**